Lab4

Kate-Shipunova opened the pull request.

ashtanyuk commented:
task4.h:3:0: warning: "SIZE" redefined
[12:55:23]	#define SIZE 220
[12:55:23]	
[12:55:23]	In file included from test_lab.cpp:2:0:
[12:55:23]	task2.h:3:0: note: this is the location of the previous definition
[12:55:23]	#define SIZE 100
[12:55:23]	
[12:55:24]	In file included from c:\googletest-master\googletest\include/gtest/gtest.h:1874:0,
[12:55:24]	from test_lab.cpp:8:
[12:55:24]	test_lab.cpp: In member function 'virtual void task3_is_palindrome_Test::TestBody()':
[12:55:24]	test_lab.cpp:36:52: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
[12:55:24]	ASSERT_EQ(0,isPalindrome("my name is Vasya"));
[12:55:24]	^
[12:55:26]	task2.cpp: In function 'char* reverseWords(char*, char*)':
[12:55:26]	task2.cpp:28:11: error: invalid conversion from 'char' to 'char*' [-fpermissive]
[12:55:26]	p[len] = '\0';// replace the last string in array 'out' for ease work
[12:55:26]

Kate-Shipunova committed:
Lab4

Kate-Shipunova committed:
Update task4.h

Kate-Shipunova committed:
Update main4.cpp

correct: 1. add type "extern" to void lineSort().
2. replaced the file name from the user with "task4.txt".

Kate-Shipunova committed:
Update task4.cpp

Kate-Shipunova committed:
Update task1.cpp

Kate-Shipunova committed:
Update task2.cpp

add: "out[j] = '\0';"

Kate-Shipunova committed:
Update main3.cpp

1. add array[]=".,!?" to replace punctuation with ' ';
2. change result=0 to result=1 if Palindrome and result=1 to result=0 if not Palindrome

Kate-Shipunova committed:
Update task3.cpp

change "return 0;" to "return 1" if Palindrome, and  "return 1;" to "return 0" if not Palindrome.
